packages/neon-core/src/tx/transaction/Transaction.ts

```diff
@@ -184,7 +183,17 @@ export class Transaction {
   }
 
   public addCosigner(...cosigners: CosignerLike[]): this {
-    this.cosigners.push(...cosigners.map(cosigner => new Cosigner(cosigner)));
+    const cosignerHashes = this.cosigners.map(cosigner => cosigner.account);
+    this.cosigners.push(
+      ...cosigners.map(cosigner => {
```

Comment:
`addCosigner(sameSig, sameSig)` will pass.

Reply:
https://github.com/CityOfZion/neon-js/blob/a30d8da56faf3c78705c5f3f84a0c6d2cead311b/packages/neon-core/__tests__/tx/transaction/Transaction.ts#L169-L183
Will not.